docs/config/ssr-options.md

```diff
@@ -20,7 +20,7 @@ Prevent listed dependencies from being externalized for SSR. If `true`, no depen
 
 ## ssr.target
 
-- **Type:** `'node' | 'webworker'`
+- **Type:** `'node' | 'webworker' | 'node-cjs'`
```

Comment:
We could also use `build.rollupOptions.output.format === 'cjs'`, but we need this target during dev for the new externalization logic. We could add a new option to switch externalization logic if not. `ssr.lazy: true`? And then people can choose to go back to the old eager collection with `ssr.lazy: false`?

Reply:
Could the externalization logic also look at `build.rollupOptions.output.format`? Having two options seems a big dangerous because someone could change build.rollupOptions.output.format, but not `ssr.target`

Reply:
Maybe we could, since we want to promote the ESM build and the new externalization logic, we don't really have a new Vite option here... it is more like a escape hatch that we're leaving for people that can yet use the ESM build. If we do that, we should remove the new `'node-cjs'` option in `ssr.target` and let users control the format and eager externalization using `build.rollupOptions.output.format: 'cjs'`.

Reply:
Same opinion as Ben: a single source of truth would be nice and `build.rollupOptions.output.format` seems like a natural fit.

Reply:
I don't think you can use `build.rollupOptions.output.format` specifically, since it may be an array?

Reply:
Yes, and there are other issues with it, like needing to conditionally change it based on the `ssr` flag. But this flag isn't currently passed to the ConfigEnv. We could add it so we could configure using `({ mode, command, ssr }) => { ... })`, but forcing a conditional as the only way to configure this doesn't seem very appealing. For other options, we discussed previously with @aleclarson and tried to avoid having SSR only equivalents, but this is a big global switch. Given that we already have `ssr.target` as an enum, I think that adding a `node-cjs` or `cjs` is ok. Although another option like `ssr.format: 'esm' | 'cjs'` could be easier to deprecate in the future... we could add it as `@experimental`, and give us room to remove it later on.

Reply:
Changed to `ssr.format: 'cjs'`, we'll later review all these new options together (there is also `ssr.scan` and `ssr.disabled: 'build'` for example)
See https://github.com/vitejs/vite/pull/8348#issuecomment-1140321430